File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        HashMap<String,Integer>map=new HashMap<>();
        for (String s: phone_book){
            map.put(s,1);
        }
        
        for(String s : map.keySet()){
            
            String temp="";
            for(int i=0;i<s.length()-1;i++){
                temp+=s.charAt(i);
                if(map.containsKey(temp))
                    return false;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    total=dict()
    for i in phone_book:
        total[i]=1
    
    
    for i in phone_book:
        temp=""
        for j in (i):
            temp+=j
            if temp in total and temp!=i:
                return False
    
    return True